lib/StaticNarrative/StaticNarrativeImpl.py

```diff
@@ -71,46 +55,19 @@ def create_static_narrative(self, ctx, params):
         """
         # ctx is the context object
         # return variables are: output
-        #BEGIN create_static_narrative
-
-        # init - get NarrativeRef and Exporter set up
-        ref = NarrativeRef.parse(params["narrative_ref"])
-        ws_url = self.config["workspace-url"]
-        self.logger.info(f"Creating Static Narrative {ref}")
-        verify_admin_privilege(ws_url, ctx["user_id"], ctx["token"], ref.wsid)
-        verify_public_narrative(ws_url, ref.wsid)
-        exporter = NarrativeExporter(self.config, ctx["user_id"], ctx["token"])
-
-        # set up output directories
-        try:
-            output_dir = os.path.join(
-                self.config["scratch"], str(ref.wsid), str(ref.objid), str(ref.ver)
-            )
-            os.makedirs(output_dir, exist_ok=True)
-        except IOError as e:
-            self.logger.error("Error while creating Static Narrative directory", e)
-            raise
-
-        # export the narrative to a file
-        try:
-            output_path = exporter.export_narrative(ref, output_dir)
-        except Exception as e:
-            self.logger.error("Error while exporting Narrative", e)
-            raise
-
-        # upload it and save it to the Workspace metadata before returning the url path
-        static_url = upload_static_narrative(ref, output_path, self.config["static-file-root"])
-        save_narrative_url(self.config["workspace-url"], ctx["token"], ref, static_url)
-        output = {
-            "static_narrative_url": static_url
-        }
-        self.logger.info(f"Finished creating Static Narrative {ref}")
-        #END create_static_narrative
+        # BEGIN create_static_narrative
+        snc = StaticNarrativeCreator(self.config)
+        params["user_id"] = ctx["user_id"]
+        params["token"] = ctx["token"]
+        output = snc.create_static_narrative(params)
```

Comment:
Everything on the left hand side has been moved with little or no change into the StaticNarrativeCreator package.